File: le1/le1-python/Exemplo1.py
import sys

# Importação fictícia, já que todas outras classes do framework ainda não existem em Python
from expression import ExpSub, ExpSoma, ValorInteiro, Programa

def main():
    # Atividade 1: 4 + 12 - 3
    exp = ExpSub(
        ExpSoma(
            ValorInteiro(4),
            ValorInteiro(12)
        ),
        ValorInteiro(3)
    )

    p = Programa(exp)
    
    if p.checaTipo():
        p.executar()
    else:
        
        print("Erro de Tipo!", file=sys.stderr)


if __name__ == "__main__":
    main() # Executando a partir da main para manter em OO
    

File: le1/le1-java/le1/plp/expressions1/Exemplo1.java
package le1.plp.expressions1;

import le1.plp.expressions1.expression.*;

public class Exemplo1 {

    public static void main(String[] args) {
        // Atividade 1: 4 + 12 - 3
        Expressao exp = new ExpSub(
            new ExpSoma(
                new ValorInteiro(4),
                new ValorInteiro(12)
            ),
            new ValorInteiro(3)
        );

        Programa p = new Programa(exp);
        if (p.checaTipo()) {
            p.executar();
        } else {
            System.err.println("Erro de Tipo!");
        }
    }
}